core/main/src/broker/thunder_broker.rs

```diff
@@ -330,3 +330,285 @@ impl EndpointBroker for ThunderBroker {
+                transform: RuleTransform::default(),
+                endpoint: None,
+                filter: None,
+            },
+            subscription_processed: None,
+        };
+
+        thndr_broker.sender.send(request).await.unwrap();
+
+        let _v = tokio::time::timeout(Duration::from_secs(2), rec.recv())
+            .await
+            .expect("Timeout while waiting for response");
+
+        assert!(_rx.recv().await.unwrap());
+
+        //invoke prepare_request()
+        let rpc_request = BrokerRequest {
+            rpc: RpcRequest::get_new_internal("some_method".to_owned(), None),
+            rule: Rule {
+                alias: "".to_owned(),
+                transform: RuleTransform::default(),
+                endpoint: None,
+                filter: None,
+            },
+            subscription_processed: None,
+        };
+
+        match thndr_broker.prepare_request(&rpc_request) {
+            Ok(requests) => {
+                for request in requests {
+                    println!("Prepared request: {}", request);
+                }
+            }
+            Err(e) => {
+                error!("Failed to prepare request: {:?}", e);
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn test_thunderbroker_get_cleaner() {
+        let (tx, mut _rx) = mpsc::channel(1);
+        let (sender, _rec) = mpsc::channel(1);
+        let send_data = vec![WSMockData::get(json!({"key":"value"}).to_string())];
+
+        let thndr_broker = get_thunderbroker(tx, send_data, sender, false).await;
+
+        let cleaner = thndr_broker.get_cleaner();
+        assert!(cleaner.cleaner.is_some());
+    }
+
+    #[tokio::test]
+    async fn test_thunderbroker_handle_jsonrpc_response() {
+        let (tx, mut _rx) = mpsc::channel(1);
+        let (sender, mut rec) = mpsc::channel(1);
+        let send_data = vec![WSMockData::get(json!({"key":"value"}).to_string())];
+
+        let _thndr_broker = get_thunderbroker(tx, send_data, sender.clone(), false).await;
+
+        let response = json!({
+            "jsonrpc": "2.0",
+            "result": {
+                "key": "value"
+            }
+        });
+
+        let callback = BrokerCallback {
+            sender: sender.clone(),
+        };
+        ThunderBroker::handle_jsonrpc_response(response.to_string().as_bytes(), callback);
+
+        let v = tokio::time::timeout(Duration::from_secs(2), rec.recv())
+            .await
+            .expect("Timeout while waiting for response");
+
+        if let Some(broker_output) = v {
+            let data = broker_output
+                .data
+                .result
+                .expect("No result in response data");
+            let key_value = data.get("key").expect("Key not found in response data");
+            let key_str = key_value.as_str().expect("Value is not a string");
+            assert_eq!(key_str, "value");
+        } else {
+            panic!("Received None instead of a valid response");
```

Comment:
change this to an assert, i.e. `assert!(broker_output.is_some())`

Reply:
review comments fixed please have a look